elasticgraph-graphql/lib/elastic_graph/graphql/filtering/filter_interpreter.rb

```diff
@@ -102,13 +201,20 @@ def filters_on_sub_fields?(expression)
           end
         end
 
+        def process_empty_or_nil_expression(bool_node, field_or_op)
+          if field_or_op == schema_names.not
+            BooleanQuery::MATCH_NONE_FILTER.merge_into(bool_node)
+          else
+            BooleanQuery::MATCH_ALL_FILTER.merge_into(bool_node)
+          end
+        end
+
         def process_not_expression(bool_node, expression, field_path)
+          # @type var sub_filter: untyped
           sub_filter = build_bool_hash do |inner_node|
             process_filter_hash(inner_node, expression, field_path)
           end
 
-          return unless sub_filter
```

Comment:
Removed this, because in theory we should never get a nil sub filter, as all emptyPredicates are treated as `match_all` and code coverage doesn't like a hanging branch. This however causes problems with steep, because `build_bool_hash` can return nil and we are trying to access `[:bool]`... So to fix this, I set sub_filter type to `untyped`. Pointing this out, incase anyone has any other paths I can take here.

Reply:
I think my suggestion about `:not` and `:empty` precedence will solve this in a more robust way.